pricegraph/src/orderbook.rs

```diff
@@ -249,26 +251,26 @@ impl Orderbook {
         }
 
         let (start, end) = (node_index(pair.buy), node_index(pair.sell));
-        let predecessors = bellman_ford::search(&self.projection, start)?.1;
-        let path = match path::find_path(&predecessors, start, end) {
+        let shortest_path_graph = bellman_ford::search(&self.projection, start)?;
+        let path = match shortest_path_graph.path_to(end) {
             Some(path) => path,
             None => return Ok(None),
         };
 
-        let flow = self.find_path_flow(&path).unwrap_or_else(|| {
+        let flow = self.find_path_flow(&path.0).unwrap_or_else(|| {
             panic!(
                 "failed to fill detected shortest path {}",
-                format_path(&path),
+                format_path(&path.0),
```

Comment:
Agree with @e00E that unpacking here is not very readable in particular here. I think a named struct would be better.

Reply:
The thing here is that `format_path` should take a path as input instead of a vector, but I didn't want to add even more changes to the code so I simply extracted the vector from the path and used the old function. Will change this!